kv/batch.go

```diff
@@ -67,7 +72,10 @@ func (rs rSpan) intersect(desc *roachpb.RangeDescriptor) rSpan {
 // overhead in the common case of a batch which never needs truncation but is
 // less magical.
 func truncate(br *roachpb.BatchRequest, desc *roachpb.RangeDescriptor, rs rSpan) (func(), int, error) {
-	rs = rs.intersect(desc)
+	rs, err := rs.intersect(desc)
+	if err != nil {
+		return func() {}, 0, nil
```

Comment:
ah. Forgot that even on error, you want to undo what was already done. The comment in `truncate()` actually mentions that.